File: rrhh-service/src/main/java/pe/albrugroup/rrhh_service/entity/response/PagoResponse.java
package pe.albrugroup.rrhh_service.entity.response;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Builder @Getter @Setter
@AllArgsConstructor @NoArgsConstructor
public class PagoResponse {

    private Long id;
    private Long idContrato;
    // PERIODO
    private LocalDate fechaInicio;
    private LocalDate fechaFin;
    // ABONO
    private BigDecimal sueldoBase;
    private BigDecimal asignacionFamiliar;
    private BigDecimal bonoPuntualidad;
    private BigDecimal comisionSemanal;
    private BigDecimal comisionMensual;
    private BigDecimal bonoExtra;
    private BigDecimal sueldoTotal;
}

File: frontend/src/app/shared/models/rrhh/pago-response.ts
export interface PagoResponse {
  id: number;
  idContrato: number;
  fechaInicio: string;
  fechaFin: string;
  sueldoBase: number;
  asignacionFamiliar: number;
  bonoPuntualidad?: number | null;
  comisionSemanal?: number | null;
  comisionMensual?: number | null;
  bonoExtra?: number | null;
  sueldoTotal: number;
  createdAt: string;
  updatedAt: string;
}
